### PR TITLE
[After #470][ml-service] Add function to transfer trained model

### PR DESCRIPTION
**Pull request by songgot**
- Add model training
- Add transfer trained model to remote training data sender after model training is completed

### CODE REVIEW

**Comment by myungjoo** [comment]
Please rebase.

**Comment by myungjoo** [comment]
Please name differently (adding ```_``` prefix) for the internal functions.
Please add "static" to in-file internal functions.
```ml_service_offloading_set_information``` may need to return an error if name is incorrect. (not supported name)

File: c/src/ml-api-service-offloading.c [hunk 1]
```diff
@@ -612,8 +599,10 @@ int
 ml_service_offloading_set_information (ml_service_h handle, const gchar * name,
```
Comment (myungjoo) [inline c/src/ml-api-service-offloading.c:613]: Is this an internal function? Is fo, add the ```_``` prefix.

File: c/src/ml-api-service-offloading.c [hunk 2]
```diff
@@ -688,6 +713,8 @@ ml_service_offloading_create (ml_service_h handle, ml_option_h option)
   return ret;
 }
 
+
+
```
Comment (myungjoo) [inline c/src/ml-api-service-offloading.c]: Meaningless lines added.

File: c/src/ml-api-service-offloading.c [hunk 3]
```diff
@@ -19,12 +19,12 @@
 #include <string.h>
 #include <curl/curl.h>
 #include <json-glib/json-glib.h>
-#include <nnstreamer-edge.h>
```
Comment (myungjoo) [inline c/src/ml-api-service-offloading.c:21]: This .C file has ```nns_edge_connect_type_e``` and ```nns_edge_node_type_e```. nnstreamer-edge.h is still valid in this file.

```
#include "ml-api-service-training-offloading.h"
```
This prevents a compiler error, but it's highly recommended to keep #include in .C file if the intefaces of ```nnstreamer-edge``` are used in this .C file directly. Headers can be kept refactored any time and ```ml-api-service-training-offloading.h``` may remove its dependency on ```nnstreamer-edge``` in the future. This .C file should work regardless of such changes in the header file.

File: c/src/ml-api-service-offloading.h [hunk 4]
```diff
@@ -14,17 +14,61 @@
...
 #endif /* __cplusplus */
 typedef void * ml_service_offloading_h;
 
+/**
+ * @brief Enumeration for ml-offloading type.
+ */
+typedef enum
+{
+  ML_OFFLOADING_TYPE_UNKNOWN = 0,
+  ML_OFFLOADING_TYPE_TRAINING,
```
Comment (myungjoo) [inline c/src/ml-api-service-offloading.h:32]: Uh.. what about the traditional inference offloading?

File: c/src/ml-api-service-offloading.h [hunk 5]
```diff
@@ -14,17 +14,61 @@
...
+  nns_edge_h edge_h;
+  nns_edge_node_type_e node_type;
+
+  gchar *path; /**< A path to save the received model file */
+  ml_option_h info;
+  GHashTable *table;
+
+  ml_offloading_type_e offloading_type;
+  void *priv;
+} _ml_service_offloading_s;
```
Comment (myungjoo) [inline c/src/ml-api-service-offloading.h:66]: Why exported?

File: c/src/ml-api-service-offloading.h [hunk 6]
```diff
@@ -14,17 +14,61 @@
...
+ * @return @c 0 on Success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #ML_ERROR_INVALID_PARAMETER Fail. The parameter is invalid.
+ * @retval #ML_ERROR_OUT_OF_MEMORY Failed to allocate required memory.
+ * @retval #ML_ERROR_STREAMS_PIPE Failed to launch the pipeline.
+ * @retval #ML_ERROR_TRY_AGAIN The pipeline is not ready yet.
+ * @retval #ML_ERROR_PERMISSION_DENIED The application does not have the privilege to access to the storage.
+ */
+int ml_service_offloading_create (ml_service_h handle);
```
Comment (myungjoo) [inline c/src/ml-api-service-offloading.h]: Do you intend to make this a public API?

File: c/src/ml-api-service-offloading.c (hunk 2 above)
Comment (songgot) [reply c/src/ml-api-service-offloading.c]: Thank you.

File: c/src/ml-api-service-offloading.h (hunk 6 above)
Comment (songgot) [reply c/src/ml-api-service-offloading.h]: No, this is removed, thank you.

File: c/src/ml-api-service-offloading.h (hunk 5 above)
Comment (songgot) [reply c/src/ml-api-service-offloading.h:66]: Yes, it will be reflected in the final commit.

File: c/src/ml-api-service-offloading.h (hunk 4 above)
Comment (songgot) [reply c/src/ml-api-service-offloading.h:32]: Yes, I will consider it. next PR.

File: c/src/ml-api-service-offloading.c (hunk 3 above)
Comment (songgot) [reply c/src/ml-api-service-offloading.c:21]: I will change this next PR

**Comment by songgot** [comment]
sqaush to https://github.com/nnstreamer/api/pull/480

**Pull request closed without merging**